lib/shopify_api/resources/application_charge.rb

```diff
@@ -2,6 +2,12 @@ module ShopifyAPI
   class ApplicationCharge < Base
     undef_method :test
 
+    class << self
+      [:pending, :cancelled, :accepted, :declined].each do |status|
+        define_method(status) { (all || []).select { |c| c.status == status.to_s } }
```

Comment:
This comparison between `c.status == status.to_s` is case sensitive. If the API ever returned statuses in a different case, it would break the API client in a subtle way.

Instead can you do something like:
`c.status.casecmp(status.to_s).zero?`  

Also can you please add a test to cover this as well?